### ISSUE TITLE
False positive: Allow trimming input before sanitizing

### ISSUE THREAD

**Issue by Luc45** [WordPress/WordPress-Coding-Standards#2180]
Sometimes, an input is flagged as not sanitized if we trim it:

```
$foo = absint( $_GET['foo'] ); // OK.

$foo = absint( trim( $_GET['foo'] ) ); // InputNotSanitized.
$foo = absint( ltrim( $_GET['foo'] ) ); // InputNotSanitized.
$foo = absint( rtrim( $_GET['foo'] ) ); // InputNotSanitized.
```

I believe trimming should be accepted before sanitizing an input.

Maybe we could add `ltrim`, `rtrim`, and `trim` as a `trimmingFunctions` array and allow them here? https://github.com/WordPress/WordPress-Coding-Standards/blob/2ac765f2b0ae653efef9f5dda74cd2324f3abe78/WordPress/Sniff.php#L1852

**Comment by dingo-d** [comment]
What error are you getting that is bothering you (I saw several when running against the `WordPress` standard), so just wanted to be sure, about the sniff you're targeting.

**Comment by Luc45** [comment]
Read the code example carefully and you will understand, it flags the usage of `trim` as if we were using the unsanitized value, but trim doesn't use the values, only manipulate it.

**Comment by jrfnl** [comment]
Just had a quick look at this, but I don't understand the use of `*trim()` here in the first place - it's not needed, so why use it ?

https://3v4l.org/9Ih5I
